tests/stubs/cloud/controllerstub/podmanager.go

```diff
@@ -0,0 +1,158 @@
+package controllerstub
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/kubeedge/beehive/pkg/common/log"
+	"github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/tests/stubs/common/constants"
+	"github.com/kubeedge/kubeedge/tests/stubs/common/types"
+	"github.com/kubeedge/kubeedge/tests/stubs/common/utils"
+)
+
+// NewPodManager creates pod manger
+func NewPodManager() (*PodManager, error) {
+	event := make(chan *model.Message)
+	pm := &PodManager{event: event}
+	return pm, nil
+}
+
+// PodManager is a manager watch pod change event
+type PodManager struct {
+	// event
+	event chan *model.Message
+	// pods map
+	pods sync.Map
+}
+
+// GetEvent return a channel which receives event
+func (pm *PodManager) GetEvent() chan *model.Message {
+	return pm.event
+}
+
+// AddPod adds pod in cache
+func (pm *PodManager) AddPod(k string, v types.FakePod) {
+	pm.pods.Store(k, v)
+}
+
+// DeletePod deletes pod in cache
+func (pm *PodManager) DeletePod(k string) {
+	pm.pods.Delete(k)
+}
+
+// UpdatePodStatus update pod status in cache
+func (pm *PodManager) UpdatePodStatus(k string, s string) {
+	v, ok := pm.pods.Load(k)
+	if ok {
+		pod := v.(types.FakePod)
+		pod.Status = s
+		pm.pods.Store(k, pod)
+	}
+}
+
+// GetPod gets pod from cache
+func (pm *PodManager) GetPod(key string) types.FakePod {
+	v, ok := pm.pods.Load(key)
+	if ok {
+		return v.(types.FakePod)
+	} else {
+		return types.FakePod{}
+	}
+}
+
+// ListPods lists all pods in cache
+func (pm *PodManager) ListPods() []types.FakePod {
+	pods := make([]types.FakePod, 0)
+	pm.pods.Range(func(k, v interface{}) bool {
+		pods = append(pods, v.(types.FakePod))
+		return true
+	})
+	return pods
+}
+
+// PodController is used to receive and process message
+func (pm *PodManager) PodController(w http.ResponseWriter, req *http.Request) {
```

Comment:
can we rename to something more meaningful like PodHandlerFunc ?

Reply:
@rohitsardesai83 Thanks. will update later